aws/retry/middleware_test.go

```diff
@@ -427,13 +430,27 @@ func TestAttemptMiddleware(t *testing.T) {
 			if diff := cmp.Diff(recorded, tt.Expect); len(diff) > 0 {
 				t.Error(diff)
 			}
-
 			attemptResults, ok := GetAttemptResults(metadata)
 			if !ok {
 				t.Fatalf("expected metadata to contain attempt results, got none")
 			}
-			if e, a := tt.ExpectResults, attemptResults; !reflect.DeepEqual(e, a) {
-				t.Fatalf("expected %v, got %v", e, a)
+			e, a := tt.ExpectResults, attemptResults
+			if len(e.Results) != len(a.Results) {
+				t.Fatalf("expected len %d got %d", len(e.Results), len(a.Results))
+			}
+			for i, eRes := range e.Results {
```

Comment:
This is a side effect of altering the return value with adding a function into it. Can no longer be deepEqualed. If anyone has a nice idea on how to fix that, I'm open for suggestions.

Sadly, it's late and I can't think of anything better ATM. :)

Reply:
I can experiment with a mock retry provider in these cases.

Reply:
We should be able to change how these values are compared to use [cmp.Diff](https://pkg.go.dev/github.com/google/go-cmp@v0.5.6/cmp#Diff). This is commonly used for other data types in the v2 SDK.

Reply:
I can try that, sure.

Reply:
Though I'm not sure how that works for functions.

Reply:
Thanks for clarifying that. Yes, you're correct cmp.Diff doens't work with functions, checking for not-nil is pretty much the only way to compare in this case.